### DISCUSSION TITLE
Nest an optimization inside a parameter study

### DISCUSSION THREAD

**Discussion by benoitpaillard** [snl-dakota/dakota#156] [Q&A]
Hi all,
I'm trying to implement a seemingly easy case, with no success so far.
I would like to optimize several operational points in a machine, which means I would set some operational parameters, then run an optimization on other parameters for each operational parameters.
I've tried adapting one of the many nested uq dakota input files, without luck so far.
Any hint would be greatly appreciated.

**Comment by thenectarine** [comment]
What is the problem statement? 

- Example 1: Minimize f(x1, x2, x3, ...) such that g(x1, x2, x3, ..) is less than 0. 
- Example 2: Minimize f(x1, x2, x3, ...) such that g(x1, x2, x3, ..) is less than 0 AND x1=2.0 and x2, x3, .. are allowed to vary.

Say you have responses ri which are dependent on variables xj, where i, j are 1, 2, 3, etc.

You want to fix one variable, e.g. x1, then optimize the other variables, e.g. x2, x3, x4, .... Then you want to repeat the optimization but by fixing x2 and varying x1, x3, x4, ....? Repeat for each variable.

Maybe this is possible with Dakota, maybe not. 

Given what I know today, I would manually create a separate Dakota input file for each optimization, then manually run each optimization and aggregate the results at the end. Alternatively, you can summon some spells with Python and automate this.
